cow_instrument/FlatTree.h

```diff
@@ -57,16 +72,37 @@ class FlatTree {
   // Needed for serialization.
   std::shared_ptr<Component> rootComponent() const;
 
+  bool operator==(const FlatTree &other) const;
+  bool operator!=(const FlatTree &other) const;
+
 private:
-  void init();
 
   /// Path index
   size_t m_sourceIndex;
   /// Path index
   size_t m_sampleIndex;
-  /// vector of proxies and relevant pointers
-  SOASource m_source;
-  /// Component root. Stashed. We only need this for serialization at present.
+
+  /*
+   These collections have the same size as the number of components. They are
+   component
+   type independent
+   */
+  std::vector<ComponentProxy> m_proxies;
+  std::vector<Eigen::Vector3d> m_positions;
+  std::vector<Eigen::Quaterniond> m_rotations;
+  std::vector<ComponentIdType> m_componentIds;
```

Comment:
@SimonHeybrock. Yes exactly. That's the crux of the changes. And with the additional constructor `LinkedTreeParser` is not always required.